root/etc/e-smith/templates/etc/httpd/conf.d/proxypass.conf/30ReverseProxyVhost

```diff
@@ -0,0 +1,26 @@
+{
+    use esmith::templates;
+    use esmith::ConfigDB;
+    my $db = esmith::ConfigDB->open_ro('proxypass') || die("Can't open proxypass db");
+
+    foreach my $proxy ($db->get_all_by_prop('type'=>'VhostReverse')) {
```

Comment:
For the reason above this loop (the whole  30ReverseProxyVhost fragment) should be moved at the bottom of `virtualhosts.conf` template

Reply:
ok you are right, we can put this fragment at the end of the virtualhost.conf, but I need to declare the default virtualhost for the 80 port  with this content, we can add more directives

```<VirtualHost _default_:80>

</VirtualHost>```

If I do not do this, all root http (80) connections are redirected to the reverse proxy, All root https connections are correctly routed to the nethserver welcome page, it is relevant only for the 80.

Fun

This why I redeclared the vhost for 443 and 80 like you saw in my 20Default_virtualhost  but it was necessary only for the 80

Reply:
You're right, we must avoid the default vhost takeover by VhostReverse. This issue is already addressed by this fragment from  `nethserver-virtualhosts`

https://github.com/NethServer/nethserver-virtualhosts/blob/1.0.5/root/etc/e-smith/templates/etc/httpd/conf.d/virtualhosts.conf/10default_vhost

We probably need to move it in nethserver-httpd

Reply:
To summarize we have the following config files (in order of priority, alphabetical)

* `nethserver.conf` -- defines the _default_:443 vhost if a TLS policy is enabled
* `proxypass.conf` -- global proxypass directives to map just URL paths, expanded by **nethserver-httpd-proxypass**
* `ssl.conf` -- upstream _default_:443
* `virtualhosts.conf` -- expanded by **nethserver-virtualhosts**, defines _default_:80

Now I think I understand better your proposal. I agree to keep both fragments under `proxypass.conf` at the following conditions:

- Remove _default_:443 from `proxypass.conf/20Default_virtualhost`. The default 443 vhost is defined by either `nethserver.conf` (TLS policy) or `ssl.conf` (upstream)
- Retain the conditional definition of _default_:80 in  `proxypass.conf/20Default_virtualhost`. Only if at least one VhostReverse is defined we expand it, so we avoid regressions on existing installations if no VhostReverse is defined.
- Remove duplicated directives for _default_:80 in `proxypass.conf/20Default_virtualhost`. Just copy the definition from [10default_vhost](https://github.com/NethServer/nethserver-virtualhosts/blob/1.0.5/root/etc/e-smith/templates/etc/httpd/conf.d/virtualhosts.conf/10default_vhost).

Reply:
To summarize we have the following config files (in order of priority, alphabetical)

* `nethserver.conf` -- defines the _default_:443 vhost if a TLS policy is enabled
* `proxypass.conf` -- global proxypass directives to map just URL paths, expanded by **nethserver-httpd-proxypass**
* `ssl.conf` -- upstream _default_:443
* `virtualhosts.conf` -- expanded by **nethserver-virtualhosts**, defines _default_:80

Now I think I understand better your proposal. I agree to keep both fragments under `proxypass.conf` at the following conditions:

- Remove _default_:443 from `proxypass.conf/20Default_virtualhost`. The default 443 vhost is defined by either `nethserver.conf` (TLS policy) or `ssl.conf` (upstream)
- Retain the conditional definition of _default_:80 in  `proxypass.conf/20Default_virtualhost`. Only if at least one VhostReverse is defined we expand it, so we avoid regressions on existing installations if no VhostReverse is defined.
- Remove duplicated directives for _default_:80 in `proxypass.conf/20Default_virtualhost`. Just copy the definition from [10default_vhost](https://github.com/NethServer/nethserver-virtualhosts/blob/1.0.5/root/etc/e-smith/templates/etc/httpd/conf.d/virtualhosts.conf/10default_vhost).